src/form/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import cache from '@dojo/framework/core/middleware/cache';
+import theme from '@dojo/framework/core/middleware/theme';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+
+import createFormMiddleware, { Validity } from './middleware';
+import * as css from '../theme/form.m.css';
+
+const form = createFormMiddleware();
+
+interface FormProperties {
+	/** The initial form value */
+	initialValue?: Record<string, any>;
+	/** Callback for when the form is submitted */
+	onSubmit(values: Record<string, any>): void;
```

Comment:
as per the example comment, do you think we need both `onSubmit` and `onValue` ?